contracts/pegged/tokens/MultiBridgePeggedToken.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity 0.8.9;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../../interfaces/IPeggedToken.sol";
+
+/**
+ * @title Example Multi-Bridge Pegged ERC20 token
+ */
+contract MultiBridgePeggedToken is IPeggedToken, ERC20, Ownable {
+    struct Supply {
+        uint256 cap;
+        uint256 total;
+    }
+    mapping(address => Supply) public bridges; // bridge address -> supply
+
+    uint8 private immutable _decimals;
+
+    event BridgeUpdated(address bridge, uint256 supplyCap);
+
+    constructor(
+        string memory name_,
+        string memory symbol_,
+        uint8 decimals_
+    ) ERC20(name_, symbol_) {
+        _decimals = decimals_;
+    }
+
+    function mint(address _to, uint256 _amount) external {
+        Supply storage b = bridges[msg.sender];
+        b.total += _amount;
+        require(b.total <= b.cap, "exceeds bridge supply cap");
+        _mint(_to, _amount);
+    }
+
+    function burn(address _from, uint256 _amount) external {
+        Supply storage b = bridges[msg.sender];
```

Comment:
Check for bridge cap > 0 first with an onlyValidBridge modifier? Otherwise technically `burn(_from, 0)` can always succeed, which is confusing.

Reply:
do you mean _amount > 0? Otherwise burn(_from, 0) can always succeed, which I think is fine, and matches the transfer behavior of most ERC20 tokens

Reply:
Ok. It's just a little bit unclear that mint / burn should only be called by the bridges, not any account.